CMake uninstall target conflicts with other previously registered uninstall targets

I am trying to build netcdf-c & netcdf-cxx4 within a cmake project via [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html). Both netcdf-c and netcdf-cxx4 register an `uninstall` which violates [CMP0002](https://cmake.org/cmake/help/latest/policy/CMP0002.html) and causes cmake to fail with this error:

```
CMake Error at build/_deps/netcdf-cxx4-src/CMakeInstallation.cmake:109 (ADD_CUSTOM_TARGET):
  ADD_CUSTOM_TARGET cannot create target "uninstall" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "build/_deps/netcdf-c-src".
  See documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  build/_deps/netcdf-cxx4-src/CMakeLists.txt:688 (INCLUDE)
```

Other projects [have solved this](https://gitlab.com/libeigen/eigen/-/issues/1842) issue by wrapping the custom target in an `option()` so that users can disable the target.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@
 # Set Project Properties
 ##################################
 
-cmake_minimum_required(VERSION 3.21.0)
+cmake_minimum_required(VERSION 3.12.0)
 
 #Project Name
 project(netCDF
@@ -15,6 +15,14 @@ HOMEPAGE_URL "https://www.unidata.ucar.edu/software/netcdf/"
 DESCRIPTION "NetCDF is a set of software libraries and machine-independent data formats that support the creation, access, and sharing of array-oriented scientific data."
 )
 set(PACKAGE "netCDF" CACHE STRING "")
+
+# Backport of built-in `PROJECT_IS_TOP_LEVEL` from CMake 3.21
+if (NOT DEFINED NETCDF_IS_TOP_LEVEL)
+  set(NETCDF_IS_TOP_LEVEL OFF)
+  if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+    set(NETCDF_IS_TOP_LEVEL ON)
+  endif ()
+endif ()
 
 #####
 # Version Info:
@@ -2787,6 +2795,8 @@ install(
 ####
 
 # CPack inclusion must come last.
-if (PROJECT_IS_TOP_LEVEL)
+option(NETCDF_PACKAGE "Create netCDF-C package " ${NETCDF_IS_TOP_LEVEL})
+
+if (NETCDF_PACKAGE)
   include(CMakeInstallation.cmake)
 endif()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,9 +6,7 @@
 # Set Project Properties
 ##################################
 
-#Minimum required CMake Version
-cmake_minimum_required(VERSION 3.12.0)
-# CMake 3.12: Use libraries specified in CMAKE_REQUIRED_LIBRARIES for check include macros
+cmake_minimum_required(VERSION 3.21.0)
 
 #Project Name
 project(netCDF
@@ -2789,5 +2787,6 @@ install(
 ####
 
 # CPack inclusion must come last.
-# INCLUDE(CPack)
-INCLUDE(CMakeInstallation.cmake)
+if (PROJECT_IS_TOP_LEVEL)
+  include(CMakeInstallation.cmake)
+endif()

